File: light_client/lib/constants/shop_catalog.dart
const Map<String, String?> shopItemPreviewImages = {
  // Characters
  'chat_astral': 'assets/images/avatars-images/avatar13.png',
  'chien_gardien': 'assets/images/avatars-images/avatar14.png',
  'lapin_lunaire': 'assets/images/avatars-images/avatar15.png',

  // Sounds
  'snd_win_music': null,
  'snd_attack': null,
  'snd_flag_music': null,

  // Visuals
  'vis_aura': null,
  'vis_flag_custom': null,
  'vis_police_nom': null,
};

String? getShopItemPreviewImage(String itemId) {
  return shopItemPreviewImages[itemId];
}

String getShopItemDisplayName(String itemId) {
  const overrides = {
    // Visuals (match server names)
    'vis_aura': 'Aura Blanche',
    'vis_flag_custom': 'Couronne Stellaire',
    'vis_police_nom': 'Nom Enchanté',
    // Sounds
    'snd_win_music': 'Musique Victoire',
    'snd_attack': 'Son Attaque',
    'snd_flag_music': 'Musique Flag',
    // Characters (match server names)
    'chat_astral': 'Chat Astral',
    'chien_gardien': 'Chien Gardien',
    'lapin_lunaire': 'Lapin Lunaire',
  };
  if (overrides.containsKey(itemId)) return overrides[itemId]!;

  final cleaned = itemId.replaceFirst(RegExp(r'^(char_|vis_|snd_|item_)'), '');
  final parts = cleaned
      .split('_')
      .map((p) => p.isEmpty ? p : '${p[0].toUpperCase()}${p.substring(1)}')
      .toList();
  return parts.join(' ');
}

String? getShopItemVisualClass(String itemId) {
  const visualClasses = {
    'vis_aura': 'visual-aura',
    'vis_flag_custom': 'visual-stars',
    'vis_police_nom': 'visual-enchanted',
  };
  return visualClasses[itemId];
}

const Map<String, String?> shopItemSounds = {
  'snd_win_music': 'assets/sounds/victory.wav',
  'snd_attack': 'assets/sounds/attack.wav',
  'snd_flag_music': 'assets/sounds/flag.wav',
};

String? getShopItemSound(String itemId) {
  return shopItemSounds[itemId];
}

File: server/app/services/currency/shop.catalog.ts
// shop.catalog.ts
export type CosmeticType = 'characters' | 'visuals' | 'sounds';

export type ShopItem = {
    id: string;
    type: CosmeticType;
    name: string;
    price: number;
};

export const SHOP_CATALOG: Record<CosmeticType, ShopItem[]> = {
    characters: [
        { id: 'chat_astral', type: 'characters', name: 'Chat Astral', price: 2 },
        { id: 'chien_gardien', type: 'characters', name: 'Chien Gardien', price: 4 },
        { id: 'lapin_lunaire', type: 'characters', name: 'Lapin Lunaire', price: 6 },
    ],
    sounds: [
        { id: 'snd_win_music', type: 'sounds', name: 'Musique Victoire', price: 8 },
        { id: 'snd_attack', type: 'sounds', name: 'Son Attaque', price: 5 },
        { id: 'snd_flag_music', type: 'sounds', name: 'Musique Flag', price: 7 },
    ],
    visuals: [
        { id: 'vis_aura', type: 'visuals', name: 'Aura Blanche', price: 9 },
        { id: 'vis_flag_custom', type: 'visuals', name: 'Couronne Stellaire', price: 8 },
        { id: 'vis_police_nom', type: 'visuals', name: 'Nom Enchanté', price: 6 },
    ],
};

export function getItem(type: CosmeticType, itemId: string): ShopItem | null {
    return SHOP_CATALOG[type]?.find((x) => x.id === itemId) ?? null;
}
